modules/andes-core/broker/src/main/java/org/wso2/andes/kernel/SetBasedExpiryMessageDeletionTask.java

```diff
@@ -0,0 +1,120 @@
+import java.util.concurrent.ExecutionException;
+import java.util.concurrent.atomic.AtomicBoolean;
+
+/**
+ * This task is responsible for delete ths expired messages captured from the message flusher
+ * Those captured messages accumulated in a set for a batch delete. This task is responsible to do
+ * that batch delete
+ */
+public class SetBasedExpiryMessageDeletionTask implements Runnable, StoreHealthListener {
+
+    /**
+     * hold the expired messages detected in the message flusher delivery rule check for batch delete
+     */
+    private static Set<DeliverableAndesMetadata> expiredMessageSet = new HashSet<>();
+
+    private static Log log = LogFactory.getLog(SetBasedExpiryMessageDeletionTask.class);
+
+    /**
+     * Indicates and provides a barrier if messages stores become offline.
+     * marked as volatile since this value could be set from a different thread
+     */
+    private volatile SettableFuture<Boolean> messageStoresUnavailable;
+
+    public SetBasedExpiryMessageDeletionTask(){
+
+        this.messageStoresUnavailable = null;
+        // Register AndesRecoveryTask class as a StoreHealthListener
+        FailureObservingStoreManager.registerStoreHealthListener(this);
+
+    }
+
+    public static Set<DeliverableAndesMetadata> getExpiredMessageSet() {
+        return expiredMessageSet;
+    }
+
+    public static void setExpiredMessageSet(Set<DeliverableAndesMetadata> expiredMessageSet) {
+        SetBasedExpiryMessageDeletionTask.expiredMessageSet = expiredMessageSet;
+    }
+
+
+    @Override
+    public void run() {
+
+        /*
+         * Checks for the message store availability if its not available
+         * Deletion task needs to await until message store becomes available
+         */
+        if (messageStoresUnavailable != null) {
+
+            try {
+                log.info("Message store has become unavailable therefore expiry message deletion task is" +
+                        "waiting until store becomes available");
+                //act as a barrier
+                messageStoresUnavailable.get();
+                log.info("Message store became available. Resuming expiry message deletion task");
+                messageStoresUnavailable = null; // we are passing the blockade
+                // (therefore clear the it).
+            } catch (InterruptedException e) {
+                log.error("Thread interrupted while waiting for message stores to come online", e);
+            } catch (ExecutionException e) {
+                log.error("Error occurred while waiting for message stores to come online", e);
+            }
+
+        }
+
+        //delete the accumulated messages in the list that are filtered out from the message flusher expiration rule
+        //should be run in all the nodes
+        if (!getExpiredMessageSet().isEmpty()) {
+            try {
+                MessagingEngine.getInstance().deleteMessages(new ArrayList<AndesMessageMetadata>(getExpiredMessageSet()));
```

Comment:
Implement a new deleteMEssages method in MessagingEngine to accept a list of messageIDs